src/box/index_def.c

```diff
@@ -168,6 +201,9 @@ index_opts_normalize(struct index_opts *opts, const struct key_def *cmp_def)
 	if (opts->layout != NULL)
 		opts->layout = xstrdup(opts->layout);
 
+	if (opts->filters != NULL)
+		opts->filters = mp_dup(opts->filters);
```

Comment:
I think this should be done in `index_def_dup` because it has nothing to do with normalization.